docs/developers/nft-tokens.md

```diff
@@ -0,0 +1,400 @@
+here in order to use it for transfers. Alternatively, the token identifier can be fetched from the API (explained also below).
+
+## **Roles** ##
+
+In order to be able to perform actions over a token, one needs to have roles assigned.
+The existing roles are:
+
+- ESDTRoleNFTCreate : this role allows one to create a new NFT/SFT
+- ESDTRoleNFTAddQuantity : this role allows one to add quantity of a specific NFT/SFT
+- ESDTRoleNFTBurn : this role allows one to burn quantity of a specific NFT/SFT
+
+To see how roles can be assigned, please here to [this](/developers/nft-tokens#assigning-roles) section.
+
+
+## **Assigning roles** ##
+
+Roles can be assigned by sending a transaction to the Metachain from the ESDT manager.
+
+Within a transaction of this kind, any number of roles can be assigned (minimum 1).
+```
+RolesAssigningTransaction {
+    Sender: <address of the ESDT manager>
+    Receiver: erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqzllls8a5w6u
+    Value: 0
+    GasLimit: 60000000
+    Data: "setSpecialRole" +
+          "@" + <token identifier in hexadecimal encoding> +
+          "@" + <address to assign the role(s) in a hexadecimal encoding> +
+          "@" + <role in hexadecimal encoding> +
+          "@" + <role in hexadecimal encoding> +
+          ...
+}
+```
+
+For example, `ESDTRoleNFTCreate` = `45534454526f6c654e4654437265617465`
+
+## **Creation of an NFT**
+
+A single address can own the role of creating an NFT for an ESDT token. This role can be transferred by using the `ESDTNFTCreateRoleTransfer` function.
+
+An NFT can be created on top of an existing ESDT by sending a transaction to self that contains the function call that triggers the creation.
+Any number of URIs can be assigned (minimum 1)
+
+```
+NFTCreationTransaction {
+    Sender: <address with ESDTRoleNFTCreate role>
+    Receiver: <same as sender>
+    Value: 0
+    GasLimit: 60000000
+    Data: "ESDTNFTCreate" +
+          "@" + <token identifier in hexadecimal encoding> +
+          "@" + <initial quantity in hexadecimal encoding> +
+          "@" + <NFT name in hexadecimal econding> +
+          "@" + <Royalties in hexadecimal encoding> +
+          "@" + <Hash in hexadecimal encoding> +
+          "@" + <Attributes in hexadecimal encoding> +
+          "@" + <URI in hexadecimal encoding> +
+          "@" + <URI in hexadecimal encoding> +
+          ...
+}
+```
+
+### **Transfer NFT Creation Role** 
+
+:::tip
+This role can be transferred only if the `canTransferNFTCreateRole` property of the token is set to `true`.
+:::
+
+The role of creating an NFT can be transferred by a Transaction like this:
+```
+TransferCreationRoleTransaction {
+    Sender: <address of the current creation role owner>
+    Receiver: erd1qqqqqqqqqqqqqqqpqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqzllls8a5w6u
+    Value: 0
+    GasLimit: 500000 + length of Data field
```

Comment:
this should be something like: 500000 + length .... in bytes * 1500? if it is computed like a normal data field.

All other methods with length should be updated.

Reply:
done